src/runtime/finalizer.cs

```diff
@@ -223,46 +169,32 @@ private void DisposeAll()
             lock (_queueLock)
 #endif
             {
+                using (Py.GIL())
```

Comment:
Why do you acquire the GIL here?

Reply:
I believe it's required for the task I'm starting at line 148, right? It can be moved inside the `Action` I'm passing to the task. It's safe to assume the thread running `Shutdown` already owns the GIL?

Reply:
I'll have to look into this in more detail, I just noticed that you added this one and that it's where the deadlock occurs on Windows :)

Reply:
Was able to repro on my end, added a commit with the fix I mentioned, all unit tests seem to be correctly passing (executed them a bunch of times). Tested with Quantconnect python tests and everything seems green there (memory leak still fixed). Let me know what you think :)